*: add more contents about supported versions and upgrade methods (#15445)
pingcap/docs: smooth-upgrade-tidb.md

> 

Reviewer: We need to remove line8-10.

Author: done

- Upgrading TiDB during the DDL operations might cause undefined behavior in TiDB.

These limitations can be summarized as that you need to ensure that there are no user-initiated DDL operations during the upgrade process. After the smooth upgrade feature is introduced, TiDB is no longer subject to this limitation during the upgrade process.

For more information, see the **Warning** content in [Upgrade TiDB Using TiUP](/upgrade-tidb-using-tiup.md#upgrade-tidb-using-tiup).

### Upgrade steps

#### Use TiUP to upgrade

Starting from v1.14.0, TiUP automatically supports this feature. That is, you can directly use the `tiup cluster upgrade` command to upgrade TiDB clusters. Note that the `tiup cluster patch` command is not supported currently.

#### Use TiDB Operator to upgrade

Currently, this feature is not supported. It will be supported as soon as possible.

#### Other upgrade methods

You can take the following steps to upgrade TiDB manually or by using a script:

1. Send the HTTP upgrade start request to any TiDB node in the cluster: `curl -X POST http://{TiDBIP}:10080/upgrade/start`.
   * The TiDB cluster enters the **Upgrading** state.

>    * The DDL operations to be performed are paused.

Reviewer: ```suggestion
   * The user DDL operations to be performed are paused.
```

- Upgrading TiDB during the DDL operations might cause undefined behavior in TiDB.

These limitations can be summarized as that you need to ensure that there are no user-initiated DDL operations during the upgrade process. After the smooth upgrade feature is introduced, TiDB is no longer subject to this limitation during the upgrade process.

### Upgrade steps

#### Use TiUP to upgrade

Starting from v1.14.0, TiUP automatically supports this feature. That is, you can directly use the `tiup cluster upgrade` command to upgrade TiDB clusters. Note that the `tiup cluster patch` command is not supported currently.

#### Use TiDB Operator to upgrade

Currently, this feature is not supported. It will be supported as soon as possible.

#### Other upgrade methods

You can use the following steps to upgrade TiDB manually or by using a script:

1. Send the HTTP upgrade start request to any TiDB node in the cluster: `curl -X POST http://{TiDBIP}:10080/upgrade/start`.
   * The TiDB cluster enters the **Upgrading** state.
   * The DDL operations to be performed are paused.

2. Replace the TiDB binary and perform a rolling upgrade. This process is the same as the original upgrade process.

>     * Perform system DDL operations during the upgrade process.

Reviewer 2: ```suggestion
    * The system DDL operations are performed during the upgrade process.
```

    * BR: BR might replicate the paused DDL jobs to TiDB. The paused DDL jobs cannot be automatically resumed, which might cause the DDL jobs to be stuck later.

    * DM and TiCDC: If you use DM or TiCDC to import SQL statements to TiDB during the upgrade process, and if one of the SQL statements contains DDL operations, the import operation is blocked and undefined errors might occur.

### Limitation on plugins

> The plugins installed in TiDB might contain DDL operations. However, during the upgrade, if the DDL operations in the plugins are not performed on system tables, the upgrade might fail.

Reviewer: ```suggestion
The plugins installed in TiDB might contain DDL operations. However, during the upgrade, if the DDL operations in the plugins are performed on non-system tables, the upgrade might fail.
```
